freud/environment.pyx

```diff
@@ -532,11 +543,27 @@ cdef class EnvironmentCluster(_MatchEnv):
                 global_search=False):
         r"""Determine clusters of particles with matching environments.
 
-        In general, it is recommended to specify a number of neighbors rather
-        than just a distance cutoff as part of your neighbor querying when
-        performing this computation. Using a distance cutoff alone could easily
-        lead to situations where a point doesn't match a cluster because a
-        required neighbor is just outside the cutoff.
+        Using a distance cutoff for :code:`'env_neighbors'` could
```

Comment:
This entire comment feels like it belongs in a `.. note::` block rather than just free-floating. This docstring could benefit from adding a short description of _how_ clusters are matched. That might be a more useful way of presenting the discussion of `neighbors` vs `env_neighbors` currently in the class docstring. Here's a suggested starting point:

> An environment is defined by the bond vectors between a particle and its neighbors as defined by `env_neighbors`. For example, `env_neighbors={'num_neighbors': 8}` means that every particle's local neighborhood is defined by its eight nearest neighbors. Then, each particle's environment is compared to the environment of the other particles in the system. Which pairs of environments are compared is controlled by the `neighbors` parameter. For example, `neighbors={'r_max': 3.0}` means that the environment of each particle will be compared to the environment of every particle within a distance of 3.0.
>
> Two environments are compared using [point-set registration](https://en.wikipedia.org/wiki/Point-set_registration). Specifically, we use the [Kabsch algorithm](https://en.wikipedia.org/wiki/Kabsch_algorithm) to find the optimal rotation matrix mapping one environment onto another. However, the Kabsch algorithm assumes a one-to-one correspondence between points in each environment, whereas we typically do not know which point in one environment corresponds to a particular point in another environment. The brute force solution here is to try all possible correspondences, but the resulting combinatorial explosion makes this problem infeasible, so we use a tolerance threshold to terminate the search when we have found a permutation of points that results in a sufficiently low RMSD.

Then you can put the "Using a distance cutoff..." part into a note.

Reply:
This is great. I added this to `EnvironmentCluster.compute`